Author: Pino Toscano <toscano.pino@tiscali.it>
Description: Fix variantIsTrue for float variants
 The test data for TestDefaultTags::testIfTag() has been expaded to
 include if-truthiness data explicitly for double and float values,
 so the they can be both tested no matter what qreal is.
Last-Update: 2013-12-08
Forwarded: https://www.gitorious.org/grantlee/grantlee/merge_requests/10

Index: grantlee/templates/tests/testdefaulttags.cpp
===================================================================
--- grantlee.orig/templates/tests/testdefaulttags.cpp	2015-05-28 11:27:27.693597227 +0200
+++ grantlee/templates/tests/testdefaulttags.cpp	2015-05-28 11:27:27.665598350 +0200
@@ -610,6 +610,20 @@
   dict.insert( QStringLiteral( "var" ), r );
   QTest::newRow( "if-truthiness10" ) << QString::fromLatin1( "{% if var %}Yes{% else %}No{% endif %}" ) << dict << QString::fromLatin1( "Yes" ) << NoError;
 
+  double d = 0.0;
+  dict.insert( QLatin1String( "var" ), d );
+  QTest::newRow( "if-truthiness11" ) << QString::fromLatin1( "{% if var %}Yes{% else %}No{% endif %}" ) << dict << QString::fromLatin1( "No" ) << NoError;
+  d = 7.1;
+  dict.insert( QLatin1String( "var" ), d );
+  QTest::newRow( "if-truthiness12" ) << QString::fromLatin1( "{% if var %}Yes{% else %}No{% endif %}" ) << dict << QString::fromLatin1( "Yes" ) << NoError;
+
+  float f = 0.0;
+  dict.insert( QLatin1String( "var" ), f );
+  QTest::newRow( "if-truthiness13" ) << QString::fromLatin1( "{% if var %}Yes{% else %}No{% endif %}" ) << dict << QString::fromLatin1( "No" ) << NoError;
+  f = 7.1;
+  dict.insert( QLatin1String( "var" ), f );
+  QTest::newRow( "if-truthiness14" ) << QString::fromLatin1( "{% if var %}Yes{% else %}No{% endif %}" ) << dict << QString::fromLatin1( "Yes" ) << NoError;
+
   dict.clear();
   QTest::newRow( "if-tag-badarg01" ) << QString::fromLatin1( "{% if x|default_if_none:y %}yes{% endif %}" ) << dict << QString() << NoError;
 
